### PR TITLE
feat(hooks): add  useExecuteOnImpression hook

### PR DESCRIPTION
**Pull request by MkMan**
Adding a hook to execute a function when the element comes into view.

hook

### CODE REVIEW

**Review by zack76 — APPROVED** [review]

**Review by hassanraza0054 — APPROVED** [review]

File: src/customHooks/useExecuteOnImpression/useExecuteOnImpression.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * A hook to execute a callback when the element enters the fold
+ * @param {HTMLElement} ref a DOM reference to the element
+ * @param {Function} functionToExecute a callback function to execute
+ * @param {{threshold: Number, shouldExecuteOnReEntry: Boolean}} options hook behaviour options
+ *
+ *
+ * @example
+ * const ref = React.useRef(null);
+ * const myFunction = () => console.log('Execute order 66');
```
Comment (beau-charman) [inline src/customHooks/useExecuteOnImpression/useExecuteOnImpression.js:13]: 😂

Comment (beau-charman) [reply src/customHooks/useExecuteOnImpression/useExecuteOnImpression.js:13]: tenor

**Review by beau-charman — CHANGES_REQUESTED** [review]
The only thing I would ask to change is the references to `functionToCall`, `functionToExecute`, etc to `callback` - to communicate what it actually is.

**Comment by beau-charman** [comment]
Like the verb `execute` is cool and all, but the use of it makes it seem that it's doing something special within this hook - like a event handler has access to its target. But this just calls a callback function. So calling the argument `callback` will communicate that. I hope that makes sense

**Review by beau-charman — APPROVED** [review]

**Pull request merged**